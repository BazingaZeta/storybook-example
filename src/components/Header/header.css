header {
  display: flex;
  margin: 0 2em;
  width: 960px;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: 'Courier New', Courier, monospace;
}
