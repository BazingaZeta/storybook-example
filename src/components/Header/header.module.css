header {
  display: flex;
  flex-flow: column;
  margin: auto 2em;
  width: 960px;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: 'Courier New', Courier, monospace;
}
