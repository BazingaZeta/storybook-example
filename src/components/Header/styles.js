import styled from 'styled-components'
import { H1 as T1, H2 as T2 } from '../typography'

export const Wrapper = styled.header`
  display: flex;
  flex-flow: column;
  padding: 0.5rem 2rem;
  margin: 0 auto;
  width: 960px;
  background-color: #ebebeb;
`

const FONT_FAMILY = "'Courier New', Courier, monospace"

export const H1 = styled(T1)`
  font-family: ${FONT_FAMILY};
`

export const H2 = styled(T2)`
  font-family: ${FONT_FAMILY};
`
