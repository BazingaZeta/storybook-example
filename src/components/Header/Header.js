import React from 'react'
import PropTypes from 'prop-types'
import { Wrapper, H1, H2 } from './styles'

const Header = ({ title, subtitle }) => (
  <Wrapper>
    <H1>{title}</H1>
    {subtitle && <H2>{subtitle}</H2>}
  </Wrapper>
)

Header.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

Header.defaultProps = {
  subtitle: null
}

export default Header
