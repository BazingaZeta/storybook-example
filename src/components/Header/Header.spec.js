import React from 'react'
import { render } from '@testing-library/react'

import Header from '.'

describe('components > Header', () => {
  const title = 'some kind of title'

  let component

  describe('renders without subtitle', () => {
    beforeAll(() => {
      component = render(<Header title={title} />)
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })

  describe('renders with subtitle', () => {
    const subtitle = 'some kind of subtitle'

    beforeAll(() => {
      component = render(<Header title={title} subtitle={subtitle} />)
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })
})
