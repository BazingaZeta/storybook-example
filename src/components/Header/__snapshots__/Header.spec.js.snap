// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Header renders with subtitle rendered without crashing 1`] = `
<DocumentFragment>
  <header
    class="sc-hKFyIo HAWcw"
  >
    <h1
      class="sc-gtssRu sc-eCApGN llUjUL koJaaW"
    >
      some kind of title
    </h1>
    <h2
      class="sc-dlnjPT sc-jSFkmK gOyEAQ inRwcu"
    >
      some kind of subtitle
    </h2>
  </header>
</DocumentFragment>
`;

exports[`components > Header renders without subtitle rendered without crashing 1`] = `
<DocumentFragment>
  <header
    class="sc-hKFyIo HAWcw"
  >
    <h1
      class="sc-gtssRu sc-eCApGN llUjUL koJaaW"
    >
      some kind of title
    </h1>
  </header>
</DocumentFragment>
`;
